plugins/catalog/src/api/CatalogClient.ts

```diff
@@ -42,4 +43,23 @@ export class CatalogClient implements CatalogApi {
     if (entity) return entity;
     throw new Error(`'Entity not found: ${name}`);
   }
+  async getLocationByEntity(entity: Entity): Promise<any> {
+    const findLocationIdInEntity = (e: Entity) => {
+      return e.metadata.annotations
+        ? e.metadata.annotations['backstage.io/managed-by-location']
+        : null;
+    };
+
+    const locationId = findLocationIdInEntity(entity);
+    if (!locationId) return null;
+
+    const response = await fetch(
+      `${this.apiOrigin}${this.basePath}/locations/${locationId}`,
+    );
+    if (response.ok) {
+      const location = await response.json();
+      if (location) return location;
+    }
+    throw new Error(`'Location not found: ${locationId}`);
```

Comment:
should it really throw? is this a fatal condition? or should it just return `undefined`

Reply:
It would be basically the case of an orphan component, right?

Reply:
Well or at least with badly formed annotations - it's not meant to be a strong foreign key but rather a helpful relation that may break sometimes. You must not make the frontend explode because of it at least :)

The only reliable source of this is the actual location-id database foreign key, and even that is not necessarily always true unless under atomicity guarantee sotuations (where you fetch all data as a single SELECT or within one transaction)

Reply:
I'm not sure. It suggests something is quite wrong since we've got an entity with a location reference that doesn't exist.

Reply:
As I feel about it, this isn't something that is going to be uncommon or catastrophic. Right now, the instant that somebody removes or mistypes something in a yaml file, things will be detached and potentially break. Do not assume that this field is perfectly reliable.

Also what will you do with the exception? Will you present it to the user somehow? What will happen to the view? What happens when you are looking at a list of hundreds of components for an entire division of the org, will one bad piece of metadata (in a component outside of your control, and not a component you were looking for) degrade your experience in the list view?

I don't want to sound negative, it's just the reality that this is weakly informative metadata, not a strict relation - at least at this point in time :)

Reply:
That makes sense. I'll treat it as nice if it works, not a big deal if it doesn't.